examples/kubernetes/README.md

```diff
@@ -1,18 +1,124 @@
-Usage
+Ceph on Kubernetes
 =====
+This Guide will take you through the process of deploying a Ceph cluster on to a Kubernetes cluster.
 
-Create a file called `ceph-mon.json` with the repository content.
-Eventually run the pod like so:
+Sigil is required for template handling and must be installed in system PATH. Instructions can be found here: [https://github.com/gliderlabs/sigil](https://github.com/gliderlabs/sigil)
+# Generate keys and configuration
 
-`kubectl create -f mon.json`
+Run the following commands to generate the required configuration and keys.
 
-Check the status of the pod:
+```
+cd generator
+./generate_secrets.sh all `./generate_secrets.sh fsid`
+
+kubectl create namespace ceph
+
+kubectl create secret generic ceph-conf-combined --from-file=ceph.conf --from-file=ceph.client.admin.keyring --from-file=ceph.mon.keyring --namespace=ceph
+kubectl create secret generic ceph-bootstrap-rgw-keyring --from-file=ceph.keyring=ceph.rgw.keyring --namespace=ceph
+kubectl create secret generic ceph-bootstrap-mds-keyring --from-file=ceph.keyring=ceph.mds.keyring --namespace=ceph
+kubectl create secret generic ceph-bootstrap-osd-keyring --from-file=ceph.keyring=ceph.osd.keyring --namespace=ceph
+kubectl create secret generic ceph-client-key --from-file=ceph-client-key --namespace=ceph
+
+cd ..
+```
```

Comment:
Yeah, I'll leave this for the tutorial but add in a script that just does everything.